source/PlayerInfo.cpp

```diff
@@ -2550,9 +2550,8 @@ void PlayerInfo::ValidateLoad()
 
 	// Validate the missions that were loaded. Active-but-invalid missions are removed from
 	// the standard mission list, effectively pausing them until necessary data is restored.
-	missions.sort([](const Mission &lhs, const Mission &rhs) noexcept -> bool { return lhs.IsValid(); });
-	auto isInvalidMission = [](const Mission &m) noexcept -> bool { return !m.IsValid(); };
-	auto mit = find_if(missions.begin(), missions.end(), isInvalidMission);
+	auto isInvalidMission = [](const Mission& m) noexcept -> bool { return !m.IsValid(); };
+	auto mit = remove_if(missions.begin(), missions.end(), isInvalidMission);
```

Comment:
Revert; this now corrupts the inactive missions, instead of preserving them for when the plugin is restored.

Reply:
How do they get corrupted?

Reply:
The only valid thing to do with elements slated for removal by `std::remove_if` is remove them or replace them with new data, as the element state is unspecified.

We put them on another list and then restore them when they are no longer invalid.

Reply:
Oh right my bad, I forgot about that. I'll use std::partition instead